application/F3DStarter.cxx

```diff
@@ -79,6 +79,27 @@ class F3DStarter::F3DInternals
       .setCurrentAsDefault();
   }
 
+  static bool HasHDRIExtension(const std::string& file)
+  {
+    std::string ext = fs::path(file).extension().string();
+    for (char& c : ext)
+    {
+      // casting is required on Windows
```

Comment:
What's the deal on Windows?  
Can't you use `std::transform`?

Reply:
toupper on windows as only `int` interface, so I need casting, and I was not able to make it work with std::transform in a oneliner. If you have a fix, Ill take it.

Reply:
Is this working?

```cpp
std::transform(str.begin(), str.end(), str.begin(), [](unsigned char c) { return std::toupper(c); });
```

Reply:
works locally, we'll see if windows is happy in CI

Reply:
```
:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\algorithm(3277,24): error C2220: the following warning is treated as an error [D:\a\f3d\f3d\build\application\f3d.vcxproj]
D:\a\f3d\f3d\source\application\F3DStarter.cxx(85,1): message : see reference to function template instantiation '_OutIt std::transform<std::_String_iterator<std::_String_val<std::_Simple_types<_Elem>>>,std::_String_iterator<std::_String_val<std::_Simple_types<_Elem>>>,F3DStarter::F3DInternals::HasHDRIExtension::<lambda_0f6ae52b19aa77bc98df59b4b693b772>>(const _InIt,const _InIt,_OutIt,_Fn)' being compiled [D:\a\f3d\f3d\build\application\f3d.vcxproj]
          with
          [
              _OutIt=std::_String_iterator<std::_String_val<std::_Simple_types<char>>>,
              _Elem=char,
              _InIt=std::_String_iterator<std::_String_val<std::_Simple_types<char>>>,
              _Fn=F3DStarter::F3DInternals::HasHDRIExtension::<lambda_0f6ae52b19aa77bc98df59b4b693b772>
          ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\algorithm(3277,24): warning C4244: '=': conversion from 'int' to 'char', possible loss of data [D:\a\f3d\f3d\build\application\f3d.vcxproj]
```

nope

Reply:
ok